src/coreComponents/constitutive/contact/BartonBandis.hpp

```diff
@@ -127,12 +137,15 @@ GEOS_HOST_DEVICE
 GEOS_FORCE_INLINE
 real64 BartonBandisUpdates::computeHydraulicAperture( real64 const aperture,
                                                       real64 const normalTraction,
+                                                      integer const fractureState,
                                                       real64 & dHydraulicAperture_aperture,
                                                       real64 & dHydraulicAperture_dNormalStress ) const
 {
-  real64 const hydraulicAperture = ( aperture >= 0.0 ) ? (aperture + m_aperture0) : m_aperture0 / ( 1 + 9*normalTraction/m_referenceNormalStress );
-  dHydraulicAperture_dNormalStress = ( aperture >= 0.0 ) ? 0.0 : -hydraulicAperture / ( 1 + 9*normalTraction/m_referenceNormalStress ) * 9/m_referenceNormalStress;
-  dHydraulicAperture_aperture = ( aperture >= 0.0 ) ? 1.0 : 0.0;
+  using namespace fields::contact;
+
+  real64 const hydraulicAperture = ( fractureState == FractureState::Open ) ? (aperture + m_aperture0) : m_aperture0 / ( 1 + 9*LvArray::math::abs(normalTraction)/m_referenceNormalStress );
+  dHydraulicAperture_dNormalStress = ( fractureState == FractureState::Open ) ? 0.0 : -hydraulicAperture / ( 1 + 9*LvArray::math::abs(normalTraction)/m_referenceNormalStress ) * 9/m_referenceNormalStress;
```

Comment:
For `fractureState != FractureState::Open`, shouldn't it be `dHydraulicAperture_dNormalStress =  -hydraulicAperture / ( 1 + 9*abs(normalTraction)/m_referenceNormalStress ) * 9/m_referenceNormalStress * normalTraction/abs(normalTraction)`?

Reply:
referenceNormalStress is enforced to be positive and an error check for input is added to avoid this confusion